Framework/PythonInterface/plugins/algorithms/SaveReflections.py

```diff
@@ -385,6 +385,53 @@ def _create_file_builders(self, file_name, workspace, split_files, scale):
 # ------------------------------------------------------------------------------------------------------
 
 
+class SHELXFormat(object):
+    """Writes a PeaksWorkspace to an ASCII file in the format required
+    by the SHELX crystallographic refinement program.
+    For constant wavelength workspaces this is an ASCII file
+    consisting of 4 columns: H, K, L, intensity, sigma.
+    For TOF Laue there are two extra columns: scaleID and wavelength.
+    """
+
+    def __call__(self, file_name, workspace, split_files, scale):
+        """
+        Write a PeaksWorkspace to an ASCII file using this formatter.
+        :param file_name: the file name to output data to.
+        :param workspace: the PeaksWorkspace to write to file.
+        :param scale: scale peaks to fit in column width
+        """
+        if has_modulated_indexing(workspace):
+            raise RuntimeError(
+                "Cannot currently save modulated structures to GSAS or SHELX formats")
+
+        with open(file_name, 'w') as f_handle:
+            self.write_peaks(f_handle, workspace, scale)
+
+    def write_peaks(self, f_handle, workspace, scale):
+        """Write all the peaks in the workspace to file.
+
+        :param f_handle: handle to the file to write to.
+        :param workspace: the PeaksWorkspace to save to file.
+        """
+        col_format = "{:>4.0f}{:>4.0f}{:>4.0f}{:>8.2f}{:>8.2f}"  # H, K, L, Intens, Sig
+        is_CW = np.std([pk.getWavelength() for pk in workspace]) < 0.01  # constant wavelength
```

Comment:
@zjmorgan @AndreiSavici - do you know of a better way to do this? I.e. could we infer that it is a constant wavelength instrument by checking if the peaks are of type `LeanPeak`, or I could use the instrument name? 
The check here I took from the algorithm `SaveHKLCW.py` (it actually appears elsewhere in `SaveReflections` - but obviously this doesn't work for a workspace with only a single peak (although that is quite unusual, it is probably more common in tests).

Reply:
That is tough to infer what is monochromatic. I think the two suggestions you point to (check wavelength tolerance or have a list of monochromatic instruments) is probably the better way to go.

In other words, constant wavelength *and* time-of-flight instruments can use `Peak` or `LeanPeak`  types and still get on. I also don't know of another attribute on any other workspace that gives that hint. In some ways, I prefer your solution. Perhaps as a longer term suggestion, we could add an attribute to the Peak objects to let the workspace know what type of instrument it is.

Reply:
> Perhaps as a longer term suggestion, we could add an attribute to the Peak objects to let the workspace know what type of instrument it is.

Thanks for the feedback, I agree long-term we should definitely have some workspace property to denote constant-wavelength data. It feels to me like this is information that should live on the `instrument` class/object, though I suppose it is possible to run some TOF instruments with constant wavelength (e.g. you could reduce a run from a direct inelastic instrument as elastic with fixed ki) - so maybe it should be on the `run` object if it depends on the instrument setup?

Reply:
> > I believe you can adaptively scale to 99999.99 and SHELX will still read it. It just expects width formatting so even the white space doesn't have to even be there. I don't particularly like that it becomes less human-readable, but it seems to work.
> 
> Yes atm it looks like the scaling is by a power of 10 such that the largest intensity has an exponent 10^3 (because it keeps the space between columns). We should probably scale the intensities such that the largest is equal to the maximum value allowed in the width which would be 99999.99 (assuming there doesn't need to be a space between columns?). Although this does effect readability... I could keep the space and scale to 9999.99? The truncation would still occur but we squeezed up to extra order of magnitude of coverage.

I think that is a good compromise. 10^5 is likely large enough.

Reply:
> > Yes atm it looks like the scaling is by a power of 10 such that the largest intensity has an exponent 10^3 (because it keeps the space between columns). We should probably scale the intensities such that the largest is equal to the maximum value allowed in the width which would be 99999.99 (assuming there doesn't need to be a space between columns?). Although this does effect readability... I could keep the space and scale to 9999.99? The truncation would still occur but we squeezed up to extra order of magnitude of coverage.
> 
> I think that is a good compromise. 10^5 is likely large enough.

OK will do. We also probably want to avoid a situation where there is a non-zero intensity and a zero error e.g.
`1   1   1    0.01    0.00   1  1.4467`
This might screw up the refinement if sigma-weights are used...